lib/rules/no-compare-relation-boolean.js

```diff
@@ -58,7 +59,15 @@ module.exports = {
             if (RELATIONAL_OPS.indexOf(binaryExprNode.operator) !== -1 && typeof literalNode.value === "boolean") {
                 context.report({
                     node: callExprNode,
-                    messageId: "redundantComparison"
+                    messageId: "redundantComparison",
+                    fix(fixer) {
+                        const sourceCode = context.getSourceCode();
+                        const assertionVariableName = callExprNode.callee.object.name;
+                        const newAssertionFunctionName = literalNode.value === !callExprNode.callee.property.name.startsWith("not") ? "ok" : "notOk"; // This is an XNOR check (checks if both or neither are true).
```

Comment:
Any way we could make this easier to read by splitting into multiple statements? The XNOR comment is helpful but I also had to concentrate to understand the precedence of the ternary operator. If anyone else needs to maintain this later, it will help if it's a bit more verbose.

Reply:
I'll try to improve the clarity here.

Reply:
Switched to a more-verbose but hopefully clearer way of writing this code.